### ISSUE TITLE
Add UI to `Custom Forms`

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver#4695]
Currently the way to create a custom form requires you to know JSON. We should add the UI so users can easier select the fields they want in the form. This would require a joined effort of devs and designers to figure out simple, yet effective interface.

![Screenshot 2023-02-23 at 1 00 29 PM](https://user-images.githubusercontent.com/36959754/220899935-0270f604-1be0-42c1-ad0d-d7af76fecad5.png)

**Comment by dstrukt** [comment]
Few Questions & Thoughts:

1. Do we still plan to support JSON input?
2. If so, would it be in addition to, or more of a binary? i.e. you choose the forms or the JSON

3. With regard to scope, are we simply create form UI for the templates for this iteration, or are we attempting to create a a flexible form builder? We touched on the idea of a form builder, but I wanted to determine involved this iteration would be, as it can get quite complex very quickly. Can design for either, just want to check with you @dennisreimann before I proceed.

4. The `Allow form for public use` switch is a little confusing in it's current form, and I think we should try to reword it to make it a little more clear.

**Comment by dennisreimann** [comment]
It will be a builder for the forms JSON, so we can look at it similar to the item/perks editor of the POS/Crowdfund apps. I'd say we still offer the "advanced mode" of editing the JSON manually (see the "raw template" option of the apps), this will also help us with debugging (at least in the first version of the new builder).

Basically we need a rather simple version to add and configure the form fields line by line. We don't need all the fancy features other form builders have, we should start with something simple but support reordering of the fields. For the option per field/line you can take a look at the [Forms documentation](https://docs.btcpayserver.org/Forms/).

The `Allow form for public use` switch enables a standalone mode, with which the form can be the entry/start for triggering an invoice. Similar to a pay button, but with additional information. I think one can also directly provide the amount. 

/cc @kukks
